src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,120 @@
+        while (scanner.hasNextLine()) {
+            String option = scanner.nextLine();
+            if (option.equals("exit")) {
+                exit();
+                break;
+            }
+            switch (option) {
+                case "add" -> addTask();
+                case "remove" -> removeTask();
+                case "list" -> displayTasks();
+                default -> System.out.println("Please select a correct option.");
+            }
+            System.out.println();
+            System.out.println("Select next option:");
+        }
+
+    }
+
+    public static void displayTasks() {
+        for (List<String> task: tasks) {
+            System.out.println(task);
+        }
+    }
+
+    public static void addTask() {
+        System.out.println("Enter the name of the task");
+        String taskName = scanner.nextLine();
+        System.out.println("Enter the due date of the task");
+        String taskDate = scanner.nextLine();
+        System.out.println("Has the task been completed? yes/no");
+        String taskStatus = scanner.nextLine();
+        while (!taskStatus.equalsIgnoreCase("yes") && !taskStatus.equalsIgnoreCase("no")) {
+            System.out.println("Please enter yes or no.");
+            taskStatus = scanner.nextLine();
+        }
+        boolean taskCompleted = taskStatus.equalsIgnoreCase("yes");
+        List<String> task = Arrays.asList(taskName, taskDate, Boolean.toString(taskCompleted));
+        tasks.add(task);
+        System.out.println("Task added: " + task);
+    }
+
+    public static void removeTask() {
+        int taskNumber;
+        System.out.println("Enter the number of the task.");
+        while (true) {
+            try {
+                String userInput = scanner.nextLine();
+                taskNumber = Integer.parseInt(userInput);
+                if (taskNumber > 0 && taskNumber <= tasks.size()) {
+                    break;
+                }
+                System.out.println("Please enter the correct value.");
+            } catch (NumberFormatException e) {
+                System.out.println("Please enter the numeric value.");
+            }
+        }
+        int taskIndex = taskNumber - 1;
+        tasks.remove(taskIndex);
+        System.out.println("Task number " + taskNumber + " removed.");
+    }
+
+    public static void exit() {
+        scanner.close();
+        Path newPath = Paths.get("tasks.csv");
+        List<String> updatedTasks = new ArrayList<>();
+        try {
+            for (List<String> task: tasks) {
+                var sb = new StringBuilder();
+                for (String word : task) {
+                    sb.append(word).append(" ");
+                }
+                updatedTasks.add(sb.toString().trim());
+            }
+            Files.write(newPath, updatedTasks);
+        } catch (IOException e) {
+            System.out.println("Unable to save the file.");
+            e.printStackTrace(System.err);
+        }
```

Comment:
CLEAN-CODE: logiku ukladani ukolu bych presunul do podmetody saveTasks().

```suggestion
        saveTasks()
```

**Navrh metody saveTasks()**
```
 private static void saveTasks() {
        final var sb = new StringBuilder();

        for (var task : tasks) {
            sb.append(String.join(", ", task));
            sb.append(System.lineSeparator());
        }

        try {
            Files.writeString(TASK_FILE_PATH, sb.toString());
        } catch (IOException e) {
            System.err.println("Failed to save tasks.csv");
            e.printStackTrace(System.err);
        }

    }
```